src/compiler/checker.ts

```diff
@@ -34454,7 +34454,7 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
     }
 
     function containerSeemsToBeEmptyDomElement(containingType: Type) {
-        return (compilerOptions.lib && !compilerOptions.lib.includes("dom")) &&
+        return (compilerOptions.lib && !compilerOptions.lib.includes("lib.dom.d.ts")) &&

```

Comment:
Do these compiler options at runtime **really** contain `lib.dom.d.ts`? That seems very weird, because in the actual settings you just provide `dom`: https://www.typescriptlang.org/tsconfig/#lib

Reply:
Yes: https://github.com/microsoft/typescript-go/pull/691#discussion_r2008826121